source/dub/internal/utils.d

```diff
@@ -609,29 +609,81 @@ string determineModuleName(BuildSettings settings, NativePath file, NativePath b
 
 /**
  * Search for module keyword in D Code
+ * A primitive parser to skip comments and whitespace to get
+ * the module's name from the module declaration.
  */
 string getModuleNameFromContent(string content) {
-	import std.regex;
-	import std.string;
+	import std.ascii: isAlpha, isAlphaNum;
+	import core.exception: RangeError;
 
-	content = content.strip;
-	if (!content.length) return null;
+	enum keyword = "module";
 
-	static bool regex_initialized = false;
-	static Regex!char comments_pattern, module_pattern;
+	size_t i = 0;
+	auto foundKeyword = false;
 
-	if (!regex_initialized) {
-		comments_pattern = regex(`//[^\r\n]*\r?\n?|/\*.*?\*/|/\+.*\+/`, "gs");
-		module_pattern = regex(`module\s+([\w\.]+)\s*;`, "g");
-		regex_initialized = true;
+	auto ch() {
+		return content[i];
 	}
 
-	content = replaceAll(content, comments_pattern, " ");
-	auto result = matchFirst(content, module_pattern);
-
-	if (!result.empty) return result[1];
+	try {
+		while(i < content.length) {
+			if(ch == keyword[0] && content[i .. i + keyword.length] == keyword) {
+				// -1 because the end of the loop will advance by 1
+				i += keyword.length - 1;
+				foundKeyword = true;
+			}
+			else if(ch == '/') {
+				++i;
+				// line comment?
+				if(ch == '/') {
+					while(ch != '\n')
+						++i;
+				}
+				// block comment?
+				else if(ch == '*') {
+					++i;
+					while(ch != '*' && content[i + 1] != '/')
+						++i;
+					++i; // skip over closing '/'
+				}
+				// nested comment?
+				else if(ch == '+') {
+					++i;
+
+					size_t level = 1;
+
+					while(level > 0) {
+						if(ch == '/') {
+							++i;
+							if(ch == '+') {
+								++i;
+								++level;
+							}
+						}
+						if(ch == '+') {
+							++i;
+							if(ch == '/') {
+								--level;
+							}
+						}
+						++i;
+					}
+				}
+			}
+			else if(isAlpha(ch) && foundKeyword) {
```

Comment:
The module name is an identifier list.

An identifier is UAX31 and is not limited to ASCII.

Instead ignore white space and look for semicolon removing the need to care about what identifier tables are in use.

Reply:
The old version had `\w` in the regex, which is the same thing as here but worse because of numbers.

Reply:
The old regex was unlikely to match what D had with the C99 tables.

Identifier tables can be interesting.

Reply:
Appears ``_`` is missing causing at least one CI failure.

So yes, this approach most definitely is missing things.

Going with a positive approach will cause problems for us long term. Going negative with what is not valid will be a lot easier as the set is rather small (whitespace and terminator ``;``).

Reply:
yeah `_` is missing, otherwise it's better than the code we had before and is

Reply:
No, it's worse.

std.regex will decode as required and compare against the alpha/numeric character tables.

This is ASCII only.

Reply:
> Going with a positive approach will cause problems for us long term. Going negative with what is not valid will be a lot easier as the set is rather small

Agreed, this should be the way to go here, code complexity and performance wise. The parser only has to work correctly under the assumption that the code is correct anyway. A sanity check for the identifier format can be done afterwards to avoid any shenanigans (e.g. injecting relative paths), but it wouldn't even matter if we had a digit in front, the compiler will already error out.